ddl_checker/main.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"database/sql"
+	"flag"
+	"fmt"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/pingcap/tidb-tools/pkg/ddl-checker"
+	"os"
+	"strings"
+	"unicode"
+)
+
+var (
+	modCode                                          = 0
+	executableChecker *ddl_checker.ExecutableChecker = nil
+	reader            *bufio.Reader                  = nil
+	db                *sql.DB                        = nil
+
+	host     = flag.String("host", "127.0.0.1", "MySQL host")
+	port     = flag.Int("port", 3306, "MySQL port")
+	username = flag.String("user", "root", "User name")
+	password = flag.String("password", "", "Password")
+	schema   = flag.String("schema", "", "Schema")
+)
+
+const (
+	WelcomeInfo = "ExecutableChecker: Check if SQL can be successfully executed by TiDB\n" +
+		"Copyright 2018 PingCAP, Inc.\n\n" +
+		"You can switch modes using the `SETMOD` command.\n" +
+		"Auto mode: The program will automatically synchronize the dependent table structure from MYSQL " +
+		"and delete the conflict table\n" +
+		"Prompt mode: The program will ask you before synchronizing the dependent table structure from MYSQL\n" +
+		"Manual mode: This program does not perform anything other than executing the input SQL.\n\n" +
+		"SETMOD usage: SETMOD <MODCODE>; MODCODE = [\"Auto\", \"Prompt\", \"Manual\"] (case insensitive).\n\n"
+)
+
+func main() {
+	initialise()
+	printWelcome()
+	mainLoop()
+	destroy()
+}
+
+func initialise() {
+	flag.Parse()
+	var err error
+	reader = bufio.NewReader(os.Stdin)
+	executableChecker, err = ddl_checker.NewExecutableChecker()
+	if err != nil {
+		fmt.Printf("[DDLChecker] Init failed, can't create ExecutableChecker: %s\n", err.Error())
+		os.Exit(1)
+	}
+	executableChecker.Execute("use test;")
+	db, err = sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:%d)/%s",
+		*username, *password, *host, *port, *schema))
+	if err != nil {
+		fmt.Printf("[DDLChecker] Init failed, can't open mysql database: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+}
+
+func destroy() {
+	executableChecker.Close()
+	db.Close()
+}
+
+func mainLoop() {
+	var input string
+	var err error
+	for cont := true; cont; cont = handler(input) {
+		fmt.Printf("[%s] > ", modeName())
+		input, err = reader.ReadString(';')
```

Comment:
can we wrap `reader.ReadString` and `strings trim and to lower` into one function? we call it at 97 and at 180